x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -574,7 +574,7 @@ export class TaskRunner<
       ILastRun
     >(
       stateWithMetrics,
-      (ruleRunStateWithMetrics) => lastRunFromState(ruleRunStateWithMetrics),
+      (ruleRunStateWithMetrics) => lastRunFromState(ruleRunStateWithMetrics, this.ruleMonitoring.getLastRun(),),
```

Comment:
Pass in current state of the `lastRun` as it might have been updated during rule execution.